code/modules/assembly/shock_kit.dm

```diff
@@ -22,7 +22,11 @@
 		electropack_part = new(src)
 		electropack_part.master = src
 
-/obj/item/assembly/shock_kit/wrench_act(mob/living/user, obj/item/I)
+/obj/item/shock_kit/receive_signal(datum/signal/signal)
+	SEND_SIGNAL(src, COMSIG_ASSEMBLY_PULSED)
```

Comment:
Signals are supposed to have a single source, not multiple.

Reply:
what do you mean?

Reply:
`COMSIG_ASSEMBLY_PULSED`, same as any other signal, should be sent with `SEND_SIGNAL` only from a single place/source. And it's already being sent by the base of `/obj/item/assembly/proc/pulsed()`, right?

In case you want to know more about the why of this, here's a handy guide:
https://tgstation13.org/phpBB/viewtopic.php?f=5&t=22674